### DISCUSSION TITLE
Official support for custom HTML Markdown anchors

### DISCUSSION THREAD

**Discussion by Marcono1234** [community/community#50962] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Is it documented somewhere whether GitHub officially supports custom HTML anchors in Markdown? For example, in some cases it might be desirable to create anchors without creating a Markdown header, or with a custom anchor ID which can be used for permalinks, regardless of whether the corresponding header is renamed later.
For example:
```md
# <a id="my-anchor"></a> Some header text which could change in the future
```

Apparently GitHub does support `<a>` with custom `id` and `name` (though `name` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#deprecated_attributes)).

The important question is however if this is officially supported (similar to the [`<details>` tag](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections)), or whether the libraries GitHub is using for Markdown rendering just happen to support this, but the behavior might change in the future.

In the past it looks like the behavior for this also changed; previously it was possible to use uppercase letters in the anchor (e.g. `<a name="TOC-Overview"></a>`), but that is not possible anymore since a few months (?)[^1]. However, all lowercase anchors (including `-`) still seem to work.

[^1]: Actually, anchors with uppercase letters still kind of work; GitHub still emits them in the HTML, but it does not scroll there anymore. Because GitHub prefixes these anchors with `user-content-` you could still use `...#user-content-TOC-Overview` but that seems to directly scroll to that HTML element, not accounting for the floating headers at the top and therefore being cut off.

**Comment by MichaelChirico** [comment]
This should be higher priority -- relying on the exact wording of anchors to generate links is extremely brittle. Any minor copy-edit has the capacity to silently break any number of downstream references.

**Comment by Marcono1234** [comment]
The GitHub documentation now describes custom anchor support, but it is not clear if this is actually a supported feature or just a description of the observed behavior from a non-GitHub employee, see https://github.com/github/docs/issues/36577.
